From 7bfc97bc5267a8ae9abb95fc4bbb2fad7f55be9c Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Fri, 17 Jun 2022 13:22:09 +0300
Subject: [PATCH] [SW-91135] gaudi2: change reset index

As part of all resets reset index for units
ARC0, ARC1, ARM CPU and PSOC has been changed
as a WA for BTM_FSM failure after hard reset.
This patch makes the change during platform
early init to make sure any reset including
prstn has correct reset index for these
components.

Change-Id: I473f17ce8c88ac7a6f54cc27ae90bc5ba84540cc
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 77039e6..bba8d2e 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -168,6 +168,14 @@
 	regmap_writel(0x0, mmPSOC_GLOBAL_CONF_BOOT_SEQ_EXT_LD);
 }
 
+void init_reset_seq(void)
+{
+	regmap_writel(0x0, mmPSOC_RESET_CONF_ARC_0_CLK_RST_CTRL);
+	regmap_writel(0x1, mmPSOC_RESET_CONF_ARC_1_CLK_RST_CTRL);
+	regmap_writel(0x2, mmPSOC_RESET_CONF_CPU_0_CLK_RST_CTRL);
+	regmap_writel(0x3, mmPSOC_RESET_CONF_PSOC_0_CLK_RST_CTRL);
+}
+
 void init_secondary(void)
 {
 	u32 rst_src;
@@ -199,6 +207,8 @@
 	/* Functionality used to be done in BTL */
 	init_prstn();
 
+	init_reset_seq();
+
 	/* If FW skip disabled update boot status that iatu is configured */
 	if (!hl_skip_fw_cfg())
 		cpu_boot_dev_sts_set(CPU_BOOT_DEV_STS_FW_IATU_CONF_EN);
